framework/src/meshgenerators/ExtraNodesetGenerator.C

```diff
@@ -78,7 +78,11 @@ ExtraNodesetGenerator::generate()
   // add nodes with their ids
   for (const auto & node_id : nodes)
```

Comment:
Can you do this instead?

```
for (const auto & node_id : nodes)
{
  const Node * node_ptr = mesh->query_node_ptr(node_id);
  if (node_ptr)
    for (const auto boundary_id : boundary_ids)
      boundary_info.add_node(node_ptr, boundary_id);
}
```

We're adding boundary_ids.size() extra node queries with what's currently there.